### PR TITLE
Deployment script Improvements

### PR DESCRIPTION
**Pull request by viraj124**
- https://goodghosting.atlassian.net/browse/VP-137
- https://goodghosting.atlassian.net/jira/software/c/projects/DEP/boards/16?modal=detail&selectedIssue=DEP-6

### CODE REVIEW

File: migrations/2_deploy_contracts.js [hunk 1]
```diff
@@ -159,7 +181,7 @@ module.exports = function (deployer, network, accounts) {
     const poolTx = await deployer.deploy(...deploymentArgs, { gasPrice: gasPrice });
     const ggInstance = await goodGhostingContract.deployed();
 
-    if (config.deployConfigs.owner && config.deployConfigs.owner != "0x") {
+    if (config.deployConfigs.owner && config.deployConfigs.owner != "0x" && config.deployConfigs.owner != "") {
```
Comment (fodisi) [inline migrations/2_deploy_contracts.js:184]: the previous condition already considered empty string (this part, `config.deployConfigs.owner`, already covers empty string and any other falsy values like `undefined`, `null`, `0`).
```suggestion
    if (config.deployConfigs.owner && config.deployConfigs.owner != "0x") {
```

File: migrations/2_deploy_contracts.js [hunk 2]
```diff
@@ -13,10 +13,27 @@ const fs = require("fs");
 const config = require("../deploy.config");
 const providerConfig = require("../providers.config");
 
+// used for amm strategies
+const tokenIndexMapping = {
```
Comment (fodisi) [inline migrations/2_deploy_contracts.js:17]: don't we need to add mobius strategies?

**Review by fodisi — CHANGES_REQUESTED** [review]

File: migrations/2_deploy_contracts.js (hunk 2 above)
Comment (viraj124) [reply migrations/2_deploy_contracts.js:17]: No we don't need there at least for the current pools

Comment (fodisi) [reply migrations/2_deploy_contracts.js:17]: you mean not for the current pools because they use 0, or because they don't use token index?

**Review by fodisi — APPROVED** [review]

File: migrations/2_deploy_contracts.js (hunk 2 above)
Comment (viraj124) [reply migrations/2_deploy_contracts.js:17]: Current pools the cusd index is 0 so hence but I'll add it going forward

**Pull request merged**